pkg/containerd/events.go

```diff
@@ -206,6 +235,91 @@ func (e *EventMonitor) Close() error {
 	return finalErr
 }
 
+// When namespaced network is enable, if portbinding is bound to 127.0.0.1 we update the DNAT
+// in main CNI DNAT chain (CNI-HOSTPORT-DNAT). The existing DNAT rule are as follows:
+// DNAT       tcp  --  anywhere             localhost            tcp dpt:9119 to:10.4.0.22:80.
+// While, this rule only allows the traffic to get routed to the localhost only we update
+// the rule to allow the traffic to any destination IP address.
+// DNAT       tcp  --  anywhere             anywhere             tcp dpt:9119 to:10.4.0.22:80.
+func execIptablesRules(portMappings nat.PortMap, containerID, namespace, pid string) error {
+	var errs []error
+
+	for portProto, portBindings := range portMappings {
+		for _, portBinding := range portBindings {
+			if portBinding.HostIP == "127.0.0.1" {
+				if err := createLoopbackIPtablesRules(
+					containerID,
+					namespace,
+					pid,
+					portProto.Port(),
+					portBinding.HostPort); err != nil {
+					errs = append(errs, err)
+				}
+			}
+		}
+	}
+
+	if len(errs) != 0 {
+		return fmt.Errorf("%w: %+v", ErrExecIptablesRule, errs)
+	}
+
+	return nil
+}
+
+func createLoopbackIPtablesRules(containerID, namespace, pid, port, destinationPort string) error {
+	// read the container's cni config to extract the network name
+	nsenterNetworkConfCmd := exec.Command("nsenter", "-t", pid, "-n", "cat", "/etc/cni/net.d/nerdctl-bridge.conflist")
```

Comment:
If we're hard-coding `nerdctl-bridge.conflist` (i.e. this is irrespective of which network the container is actually in), couldn't we have hard-coded `bridge` here too? Or am I missing something here (likely)?

Reply:
Created an issue: https://github.com/rancher-sandbox/rancher-desktop-agent/issues/47